### DISCUSSION TITLE
After freeing a world and recreating it, objects seem to keep their momentum

### DISCUSSION THREAD

**Discussion by mucookul** [dimforge/rapier#575] [Q&A] [answered]
After clicking the reset button multiple times, you can start to notice the cube fall faster than it should.
My project: https://stackblitz.com/edit/js-j1lsjg

Thanks

**Comment by doppl3r** [comment, marked as answer]
You'll want to remove the `requestAnimationFrame(() => gameLoop(scalingFactors));` line within your `resetButton` click event. You are currently doubling your `gameLoop` function every time you click the reset button.

**Comment by mucookul** [comment]
I though I would catch a issue like that. Thanks!
